Add beacon info to app

danielcwilson opened the pull request:
Needs testing; should have added a tab with the old info screen that we had alongside the map screen
Includes horrible implementation of many ifs to try handle signal loss and stuff as well

danielcwilson committed:
Inital merge but whoops forgot Alexs new stuff

Will fix I promise

danielcwilson committed:
Merge "Show map with buttons that change color"

This probably works but i can't test it, also not finished the reset button in the debug screen yet

danielcwilson committed:
Fix "connections" whatever they are

Turns out you can't just copy/paste with xcode or else it gets real angry

danielcwilson committed:
Possibly set up reset button

I see the possibility for many issues with this and also I still hate myself for the nasty if block.
Pls test

vaguelycompetent commented:
I don't know what the merge issues are cause I had to merge myself earlier anyway, should just take all of my changes

danielcwilson committed:
Attempt to fix the unexpected nil || Change debug logic

It was actually expected because I set it to nil but apparently swift doesn't like that.
I also realised that my long horrible chain of ifs was wrong so I changed the logic there. Lets hope for at least not the worst

vaguelycompetent commented:
Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArray0 objectAtIndex:]: index 0 beyond bounds for empty NSArray'

whoops okay

danielcwilson committed:
Fix the nil thing

I'm actually furious that this is what it took to fix it

Now that I can see whats going on, the compass and map rotation both seem to be broken

danielcwilson committed:
Fix rotation of map and compass

Both now go the right way, buuuut I found out I drew the map wrong but nobody noticed; that'll get fixed soon probably

vaguelycompetent commented:
Okay so I can't test with beacons but I think this is good now. What I did found out eventually was that the map has been drawn wrong, the office is on the wrong side of the elevator room. I have way less of an idea of the shape of this building than I thought.

Everything will look wrong for now but it works.

danielcwilson committed:
Add some nice comments and tidy up a little

Now if no beacons are found the info page will tell you as well

danielcwilson committed:
Stop crashing on only one beacon detected

This only stops crashes, it won't actually show anything for the one beacon

vaguelycompetent commented:
Going in a different direction with this eventually
